### PR TITLE
Fix Definite Integral for functions with ConditionSet singularities

### LINKED ISSUE #15285: TypeError: 'ConditionSet' object is not iterable when calculating definite integral
```
>>> from sympy import *
>>> from sympy.abc import x,y
>>> f = (4 * (1/y - 1) * exp(-2 * (1/y - 1))) / (y**2)
>>> integrate(f,(y,0,1))
...
    for s in singularities:
TypeError: 'ConditionSet' object is not iterable
```

A similar issue happened in issue #13433, though this was fixed inadvertently by the changes in #14097. In this instance, the singularities of the function are defined by the ConditionSet

```
ConditionSet(y, Eq(y*exp(2/y), 0), Interval(0, 1))
```

and so when trying to iterate over this, python throws an error. I think an easy fix would be to catch the exception and pass, but there could be a better solution.

### PR DESCRIPTION
**Pull request by apalmer1377**
#### References to other Issues or PRs

Fixes #15285

#### Brief description of what is fixed or changed
Certain functions cause evaluating a definite integral to fail, due
to the set of singularities of the function being a ConditionSet
(and therefore not iterable). The change just catches the TypeError
and passes it.

#### Release Notes

* core
    * fixed bug that occurred when integrating certain functions over an interval

### CODE REVIEW

**Comment by asmeurer** [comment]
This needs a test.

**Comment by apalmer1377** [comment]
Added a test to check validity of fix.

File: sympy/integrals/tests/test_integrals.py [hunk 1]
```diff
@@ -1388,3 +1388,8 @@ def test_issue_14782():
 def test_issue_12081():
     f = x**(-S(3)/2)*exp(-x)
     assert integrate(f, [x, 0, oo]) == oo
+
+def test_issue_15285():
+    y = 1/x - 1
+    f = (4 * y * exp(-2 * y)) / (x**2)
+    assert(integrate(f,[x,0,1]) == 1)
```
Comment (sidhantnagpal) [inline sympy/integrals/tests/test_integrals.py]: A few formatting issues to take care of:
- Methods are separated using 2 newline characters
- See usage for spaces around operators, `f = (4*y*exp(-2*y))/(x**2)`
- Spaces after commas, `assert integrate(f, [x, 0, 1]) == 1`

File: sympy/integrals/tests/test_integrals.py [hunk 2]
```diff
@@ -1388,3 +1388,9 @@ def test_issue_14782():
 def test_issue_12081():
     f = x**(-S(3)/2)*exp(-x)
     assert integrate(f, [x, 0, oo]) == oo
+
+
+def test_issue_15285():
+    y = 1/x - 1
+    f = (4*y*exp(-2*y))/(x**2)
```
Comment (Abdullahjavednesar) [inline sympy/integrals/tests/test_integrals.py]: Just `f=4*y*exp(-2*y)/x**2` should suffice.

Comment (jksuom) [reply sympy/integrals/tests/test_integrals.py]: with a space on either side of `=`.

**Comment by sidhantnagpal** [comment]
This looks fine and the tests pass (the job 21 failed earlier due to HTTP error).

**Pull request merged**